datadog_checks_dev/datadog_checks/dev/tooling/commands/validate/ci.py

```diff
@@ -178,3 +193,155 @@ def ci(ctx, fix):
         output = yaml.safe_dump(jobs_definition, default_flow_style=False, sort_keys=False)
         write_file(jobs_definition_path, output)
         echo_success('Successfully fixed {}'.format(jobs_definition_relative_path))
+
+    codecov_config_relative_path = repo_data['codecov_config_relative_path']
+    codecov_config_path = path_join(root, *codecov_config_relative_path.split('/'))
+    if not file_exists(codecov_config_path):
+        abort('Unable to find the Codecov config file')
+
+    codecov_config = yaml.safe_load(read_file(codecov_config_path))
+    projects = codecov_config.setdefault('coverage', {}).setdefault('status', {}).setdefault('project', {})
+
+    defined_checks = set()
+    success = True
+    fixed = False
+
+    for project, data in list(projects.items()):
```

Comment:
This function is already really large - can we refactor this out to a separate function with a docstring describing what it's doing?